### PR TITLE
Disallow contests with multiple timezones

### PR DESCRIPTION
**Pull request by vmcj**
Closes https://github.com/DOMjudge/domjudge/issues/1134

### CODE REVIEW

File: webapp/src/Controller/Jury/ContestController.php [hunk 1]
```diff
@@ -492,6 +492,21 @@ public function editAction(Request $request, int $contestId): Response
...
         if ($form->isSubmitted() && $form->isValid()) {
+            $formData = $form->getData();
+            $timeZones = [];
+            foreach(['Activate','Deactivate','Start','End','Freeze','Unfreeze'] as $timeString) {
+                $fields = explode(' ', $formData->{'get'.$timeString.'timeString'}());
+                if(count($fields)>1) {
+                    $timeZones[] = $fields[2];
+                }
+            }
+            if(count(array_unique($timeZones))>1) {
```
Comment (nickygerritsen) [inline webapp/src/Controller/Jury/ContestController.php:503]: Could we try to format this according to the rest of the file (PSR)?

**Review by nickygerritsen — APPROVED** [review]
Looks good

File: webapp/src/Controller/Jury/ContestController.php (hunk 1 above)
Comment (vmcj) [reply webapp/src/Controller/Jury/ContestController.php:503]:
```suggestion
            if (count(array_unique($timeZones))>1) {
```
So this?

Comment (nickygerritsen) [reply webapp/src/Controller/Jury/ContestController.php:503]: No, even `if (count(array_unique($timeZones)) > 1) {`

File: webapp/src/Controller/Jury/ContestController.php [hunk 2]
```diff
@@ -492,6 +492,21 @@ public function editAction(Request $request, int $contestId): Response
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
+            $formData = $form->getData();
+            $timeZones = [];
+            foreach (['Activate','Deactivate','Start','End','Freeze','Unfreeze'] as $timeString) {
+                $fields = explode(' ', $formData->{'get'.$timeString.'timeString'}());
+                if (count($fields)>1) {
```
Comment (nickygerritsen) [inline webapp/src/Controller/Jury/ContestController.php:499]:
```suggestion
                if (count($fields) > 1) {
```

File: webapp/src/Controller/Jury/ContestController.php [hunk 3]
```diff
@@ -492,6 +492,21 @@ public function editAction(Request $request, int $contestId): Response
...
+                    $timeZones[] = $fields[2];
+                }
+            }
+            if (count(array_unique($timeZones))>1) {
```
Comment (nickygerritsen) [inline webapp/src/Controller/Jury/ContestController.php:503]:
```suggestion
            if (count(array_unique($timeZones)) > 1) {
```

**Comment by vmcj** [comment]
I've added some extra tests, @nickygerritsen do you still approve?

**Review by nickygerritsen — APPROVED** [review]
Yes

**Pull request merged**